### ISSUE TITLE
Client/Server Multisig Setup with Beet

### ISSUE THREAD

**Issue by xeroc** [bitshares/beet#69]
**Is your feature request related to a problem? Please describe.**
I want to be able to setup multisig system that consists of a client (beet) and a server. Both need to sign a transaction.

**Describe the solution you'd like**
The process I have in mind looks like this:
1. The server creates a raw (unsigned transaction)
2. Sever sends it to the client
3. Clients signs the raw transaction (one out of two require signatures) using beet
4. Client sends the partially signed tx to server
5. Server adds the second signature and broadcasts the transaction

**Describe alternatives you've considered**
The alternative would be to do it the other way round and let the server sign first. The difference should be minimal as you just have to keep track of a potentially existing signature in the to-be-signed transaction.

**Additional context**
We would like to be able to improve security by removing the potential of the server-installed private key to do anything without the approval of a local signature.

**Comment by clockworkgr** [comment]
Do you want an off-chain synchronous solution (which is what I understand from the flow description) without using proposals?

Or would something like Beet monitoring the saved accounts for pending proposals and notify/prompt the user for approval suffice?

**Comment by xeroc** [comment]
> Do you want an off-chain synchronous solution (which is what I understand from the flow description) without using proposals?

Yes. A flow that ends up in a transaction with two signatures. One fro beet, one from a dedicated server



> Or would something like Beet monitoring the saved accounts for pending proposals and notify/prompt the user for approval suffice?

Nop. I don't want to deal with proposals at all in my use-case.

**Comment by clockworkgr** [comment]
Hmmm...this would possibly require exposing the Beet websocket port to the internet

**Comment by clockworkgr** [comment]
that is not a problem really...it could be left up to the user to proxy the port and secure it...the protocol itself is quite robust for MITM attacks...

but I guess you need a python client library for it :)

**Comment by xeroc** [comment]
> Hmmm...this would possibly require exposing the Beet websocket port to the internet

The process would be like this:
1. Server provides unsigned tx via javascript to a webpage
2. Webpage has beet integration and requests signature for unsigned tx
3. partially signed tx is pushed to server
4. server adds his signature (e.g. cli-wallet, pybitshares, ...)
5. server broadcasts tx

**Comment by clockworkgr** [comment]
Ah...simpler...albeit convoluted...thought you wanted to automate the first part

**Comment by xeroc** [comment]
> Ah...simpler...albeit convoluted...thought you wanted to automate the first part

The application here is `workers.bitshares.exchange`. I want to setup multisig there so that hacks into the server cannot compromise funds without also hacking someone else. Also, I don't want the *someone else* to be able to move funds. Hence multisig. 
Additionally, I need the server to broadcast the transaction for reasons of accounting (in case a order is filled, or a tx fails).
Basically, I want the user to confirm a transaction with beet and have the server add his signature prior to broadcasting.

**Comment by clockworkgr** [comment]
yep...gotcha... on it
